base/range.jl

```diff
@@ -1388,9 +1388,7 @@ function sum(r::AbstractRange{<:Real})
 end
 
 function _in_range(x, r::AbstractRange)
-    if !isfinite(x)
-        return false
-    elseif iszero(step(r))
+    if iszero(step(r))
```

Comment:
```suggestion
    isnan(x) && return false
    if iszero(step(r))
```
The `isfinite` check implicitly covered NaNs. I think an explicit check for NaN is necessary if it's removed? Otherwise `NaN in 1.0:2.0` would throw instead of returning `false`.

Reply:
Good catch! Unfortunately, that fix will not work because we don't have `isnan` for non-numeric ranges.

Reply:
I used `first(r) <= x <= last(r)` instead.